test/comparisons.jl

```diff
@@ -160,6 +168,8 @@ const INTERVAL_TYPES = [Interval, AnchoredInterval{Ending}, AnchoredInterval{Beg
         @test intersect(earlier, later) == Interval(3, 3, true, true)
         @test merge(earlier, later) == Interval(1, 5, true, true)
         @test union([earlier, later]) == [Interval(1, 5, true, true)]
+        @test overlaps(earlier, later)
+        @test contiguous(earlier, later)
```

Comment:
Hmmm, I think `contiguous` should be `false` here

Reply:
This brings a new test to mind. For intervals using discrete units like integers the intervals `[1,2]` and `[3,4]` should be contiguous. I'll make that a separate issue.

Reply:
Issue I mentioned: https://github.com/invenia/Intervals.jl/issues/36